apps/extension/src/core/views/Settings.tsx

```diff
@@ -65,8 +60,14 @@ export function Settings() {
     })
   }
 
-  const isLocalModel = modelId === ModelID.Local
-  const isOpenAI = modelId === ModelID.GPT3 || modelId === ModelID.GPT4
+  const isLocalModel =
+    config?.auth === AuthType.APIKey && config?.models.length === 0
+  const needsAPIKey = config?.auth === AuthType.APIKey
+  const asksForAPIKey = needsAPIKey || isLocalModel // Some local models need keys, e.g. https://github.com/keldenl/gpt-llama.cpp
+  const isExternal = config?.auth === AuthType.External
+  const isOpenAIAPI =
+    config?.auth === AuthType.APIKey &&
+    !!config?.models.find((m) => m === ModelID.GPT3 || m === ModelID.GPT4)
```

Comment:
nit - these checks look complex enough to useMemo:

```
const {
    isLocalModel 
} = useMemo(() => {

  const isLocalModel =
    config?.auth === AuthType.APIKey && config?.models.length === 0
  const needsAPIKey = config?.auth === AuthType.APIKey
  const asksForAPIKey = needsAPIKey || isLocalModel // Some local models need keys, e.g. https://github.com/keldenl/gpt-llama.cpp
  const isExternal = config?.auth === AuthType.External
  const isOpenAIAPI =
    needsAPIKey  &&
    !!config?.models.find((m) => m === ModelID.GPT3 || m === ModelID.GPT4)
    
    return {
    isLocalModel ,
    etc...
    }

}, [config])
```

Reply:
nice call